testutils/check-lang-uses-form/main.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode/utf8"
+)
+
+const usage = `Reads calling forms from -forms and makes sure each is used by
+the specified languages' *.baseline file in dir.
+The line specifying the calling forms is in the format:
+
+  CallingForm[,] // used by: lang1 lang2 ...
+
+If the string "// used by" is itself preceeded by another "//" in the same line,
+it is ignored so that it is possible to talk about the comment in the file itself.
+
+Baseline files of the specified languages (in this case "lang1" and "lang2") will be checked
+`
+
+func main() {
+	formFname := flag.String("forms", "", "file containing calling forms")
+	flag.Usage = func() {
+		binName := os.Args[0]
+		out := os.Stderr // Should be flag.CommandLine.Output(), but Go version on Travis is too old.
+		fmt.Fprintf(out, "Usage of %s:\n", binName)
+		fmt.Fprintf(out, "%s -form CallingForm.java [dir]\n", binName)
+		flag.PrintDefaults()
+		fmt.Fprint(out, usage)
+	}
+	flag.Parse()
+
+	if *formFname == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
```

Comment:
I would recommend making main deal only with parsing arguments and printing usage, and have it pass the arguments to a function that does the logic based on its formal parameters. At the moment, it's surprising that you look at `Nargs` further down in this function, for example.

Reply:
Done